src/Microsoft.TemplateEngine.Cli/NewCommandFactory.cs

```diff
@@ -4,14 +4,15 @@
 #nullable enable
 
 using System.CommandLine;
+using System.CommandLine.Parsing;
 using Microsoft.TemplateEngine.Abstractions;
 using Microsoft.TemplateEngine.Cli.Commands;
 
 namespace Microsoft.TemplateEngine.Cli
 {
     public static class NewCommandFactory
     {
-        public static Command Create(string commandName, ITemplateEngineHost host, ITelemetryLogger telemetryLogger, NewCommandCallbacks callbacks)
+        public static Command Create(string commandName, Func<ParseResult, ITemplateEngineHost> host, Func<ParseResult, ITelemetryLogger> telemetryLogger, NewCommandCallbacks callbacks)
```

Comment:
Nit: It would be nice to rename the parameters as well. Something like `hostSelector`.